test/object/object.cc

```diff
@@ -83,20 +92,36 @@ void DefineProperties(const CallbackInfo& info) {
     std::string str7("function");
 
     obj.DefineProperties({
+#ifndef NODE_ADDON_API_DISABLE_DEPRECATED
+      PropertyDescriptor::Accessor(str1, TestGetter),
+      PropertyDescriptor::Accessor(str2, TestGetter, TestSetter),
+#else // NODE_ADDON_API_DISABLE_DEPRECATED
       PropertyDescriptor::Accessor(env, obj, str1, TestGetter),
       PropertyDescriptor::Accessor(env, obj, str2, TestGetter, TestSetter),
+#endif // !NODE_ADDON_API_DISABLE_DEPRECATED
       PropertyDescriptor::Value(str3, trueValue),
       PropertyDescriptor::Value(str4, trueValue, napi_writable),
       PropertyDescriptor::Value(str5, trueValue, napi_enumerable),
       PropertyDescriptor::Value(str6, trueValue, napi_configurable),
+#ifndef NODE_ADDON_API_DISABLE_DEPRECATED
+      PropertyDescriptor::Function(str7, TestFunction),
+#else // NODE_ADDON_API_DISABLE_DEPRECATED
       PropertyDescriptor::Function(env, str7, TestFunction),
+#endif // !NODE_ADDON_API_DISABLE_DEPRECATED
```

Comment:
I think we want to test both the deprecated and non-deprecated so this should just be #ifndef, #endif as opposed to using #else ?  Same goes to all of the other places as well.